### PR TITLE
Refactors

### PR DESCRIPTION
**Pull request by curiousleo**
A few automated refactors. The commits are self-contained.

Feel free to ignore - some of these are stylistic questions about which opinions can differ!

### CODE REVIEW

**Comment by draeger** [comment]
My only concern with the suggested changes would be if they enforce an upgrade in the end-users' Java™ version? We need to carefully consider if we loose users of SBSCL by enforcing higher Java versions or if it is backward compatible? Which would be the minimal Java version that can still be used after these changes?

File: src/main/java/fern/analysis/NetworkSearchAction.java [hunk 1]
```diff
@@ -20,28 +20,28 @@
    *
    * @param reaction index of the inserted reaction
    */
-  public void reactionDiscovered(int reaction);
```
Comment (shalinshah1993) [inline src/main/java/fern/analysis/NetworkSearchAction.java:23]: What's the argument for deleting `public` here? It is nice if we have explicit mention.

File: src/main/java/fern/analysis/ShortestPath.java [hunk 2]
```diff
@@ -202,7 +202,7 @@ public ShortestPathAction(NodeChecker checker) {
      * @return
      */
     public Path getPath(int species) {
-      LinkedList<Integer> path = new LinkedList<Integer>();
```
Comment (shalinshah1993) [inline src/main/java/fern/analysis/ShortestPath.java:205]: I like the explicit mention type when initializing. Makes things little easy to follow. I'd prefer keeping this the original way.

**Review by draeger — COMMENTED** [review]
@curiousleo: My only concern with the suggested changes would be if they enforce an upgrade in the end-users' Java™ version? We need to carefully consider if we loose users of SBSCL by enforcing higher Java versions or if it is backward compatible? Which would be the minimal Java version that can still be used after these changes?

File: src/main/java/fern/analysis/NetworkSearchAction.java (hunk 1 above)
Comment (curiousleo) [reply src/main/java/fern/analysis/NetworkSearchAction.java:23]: https://docs.oracle.com/javase/tutorial/java/IandI/interfaceDef.html

> All abstract, default, and static methods in an interface are implicitly public, so you can omit the public modifier.

> All constant values defined in an interface are implicitly public, static, and final. Once again, you can omit these modifiers.

Declaring an interface method `public` carries the same amount of information as `/* this is a comment */` :)

File: src/main/java/fern/analysis/ShortestPath.java (hunk 2 above)
Comment (curiousleo) [reply src/main/java/fern/analysis/ShortestPath.java:205]: Fair! I personally find Java verbose enough with all the _necessary_ declaration, so I tend to trim the unnecessary ones as much as possible - it seems clear here even in the less explicitly typed version that a `LinkedList` containing `Integer`s is being created. But tastes differ!

**Comment by curiousleo** [comment]
> @curiousleo: My only concern with the suggested changes would be if they enforce an upgrade in the end-users' Java™ version? We need to carefully consider if we loose users of SBSCL by enforcing higher Java versions or if it is backward compatible? Which would be the minimal Java version that can still be used after these changes?

Fair point!

Changing `jdk.version` in `pom.xml` and then running `mvn compile` shows that `master` as of 7d9548820719b587f36c54157537700afd40dbec

- Fails to build against JDK 6: "diamond operator is not supported"
- Fails to build against JDK 7: "lambda expressions are not supported"
- Successfully builds against JDK 8 (that's what the default is in `pom.xml` and what runs in CI)

This PR also builds against JDK 8 (if not, it would have failed CI). So there's no regression in terms of which JDK versions SBSCL can build against, compatibility is unchanged.

**Comment by draeger** [comment]
> > @curiousleo: My only concern with the suggested changes would be if they enforce an upgrade in the end-users' Java™ version? We need to carefully consider if we loose users of SBSCL by enforcing higher Java versions or if it is backward compatible? Which would be the minimal Java version that can still be used after these changes?
> 
> Fair point!
> 
> Changing `jdk.version` in `pom.xml` and then running `mvn compile` shows that `master` as of [7d95488](https://github.com/draeger-lab/SBSCL/commit/7d9548820719b587f36c54157537700afd40dbec)
> 
> * Fails to build against JDK 6: "diamond operator is not supported"
> * Fails to build against JDK 7: "lambda expressions are not supported"
> * Successfully builds against JDK 8 (that's what the default is in `pom.xml` and what runs in CI)
> 
> This PR also builds against JDK 8 (if not, it would have failed CI). So there's no regression in terms of which JDK versions SBSCL can build against, compatibility is unchanged.

Ok, I think, Java 8 is nearly 7 years old and, therefore, a requirement we can assume users to have installed by now.

**Pull request merged**